lib/src/Base/Stat/RankMCovarianceModel.cxx

```diff
@@ -114,27 +114,23 @@ CovarianceMatrix RankMCovarianceModel::operator() (const Point & s,
     }
   else
   {
-    MatrixImplementation phiS(outputDimension_, size);
     MatrixImplementation phiT(outputDimension_, size);
-    Collection<Scalar>::iterator itPhiS = phiS.begin();
     Collection<Scalar>::iterator itPhiT = phiT.begin();
     for (UnsignedInteger i = 0; i < size; ++i)
     {
-      const Point evalPhiS(functions_[i](s));
-      std::copy(evalPhiS.begin(), evalPhiS.end(), itPhiS);
-      itPhiS += outputDimension_;
       const Point evalPhiT(functions_[i](t));
       std::copy(evalPhiT.begin(), evalPhiT.end(), itPhiT);
       itPhiT += outputDimension_;
     }
     for (UnsignedInteger i = 0; i < size; ++i)
     {
+      const Point phiS(functions_[i](s));
       itPhiT = phiT.begin();
       for (UnsignedInteger j = 0; j < size; ++j)
       {
         const Point ptPhiT(Point(Collection<Scalar>(itPhiT, itPhiT + outputDimension_)) * covariance_(i, j));
         dger_(&dim, &dim, &plusOne,
```

Comment:
Not very fond of direct calls to LAPACK, but when the speed-up is significant and there is no other option...